aws/resource_aws_lambda_permission_test.go

```diff
@@ -315,6 +316,37 @@ func TestAccAWSLambdaPermission_withQualifier(t *testing.T) {
 	})
 }
 
+func TestAccAWSLambdaPermission_disappears(t *testing.T) {
+	var statement LambdaPolicyStatement
+
+	rString := acctest.RandString(8)
+	funcName := fmt.Sprintf("tf_acc_lambda_perm_basic_%s", rString)
+	roleName := fmt.Sprintf("tf_acc_role_lambda_perm_basic_%s", rString)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLambdaPermissionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLambdaPermissionConfig(funcName, roleName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLambdaPermissionExists("aws_lambda_permission.allow_cloudwatch", &statement),
+				),
+			},
+			// Here we delete the Lambda permission to verify the follow-on refresh after this step
+			// should not error.
+			{
+				Config: testAccAWSLambdaPermissionConfig(funcName, roleName),
```

Comment:
The error only shows up when there are still permission elements left. We can use `testAccAWSLambdaPermissionConfig_multiplePerms()` instead, and then remove `aws_lambda_permission.first`.